homeassistant/components/fibaro/__init__.py

```diff
@@ -305,6 +318,44 @@ def _map_device_to_platform(device: Any) -> Platform | None:
             platform = Platform.LIGHT
         return platform
 
+    def _create_device_info(self, device: Any, devices: list) -> None:
+        """Create the device info. Unrooted entities are directly shown below the home center."""
+
+        # The home center is always id 1 (z-wave primary controller)
+        if "parentId" not in device or device.parentId <= 1:
+            return
+
+        master_entity: Any | None = None
+        if device.parentId == 1:
+            master_entity = device
+        else:
+            for parent in devices:
+                if "id" in parent and parent.id == device.parentId:
+                    master_entity = parent
+        if master_entity is None:
+            _LOGGER.error("Parent with id %s not found", device.parentId)
+            return
+
+        if "zwaveCompany" in master_entity.properties:
+            manufacturer = master_entity.properties.zwaveCompany
+        else:
+            manufacturer = "Unknown"
+
+        self._device_infos[master_entity.id] = DeviceInfo(
+            identifiers={(DOMAIN, master_entity.id)},
+            manufacturer=manufacturer,
+            name=master_entity.name,
+            via_device=(DOMAIN, str(self.hub_device_info[ATTR_MODEL])),
```

Comment:
This does not seem correct. The via_device should have the same identifier as the hub, which above is set to `{(DOMAIN, info.serialNumber)}`